Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.API/Controllers/SyncKronosToShiftsController.cs

```diff
@@ -104,51 +104,122 @@ public ActionResult SyncKronosToShifts(string isRequestFromLogicApp)
         /// <returns>Returns task.</returns>
         private async Task ProcessKronosToShiftsShiftsAsync(string isRequestFromLogicApp)
         {
-            try
+             var isOpenShiftRequestSyncSuccessful = false;
+             var isSwapShiftRequestSyncSuccessful = false;
+             var isMapPayCodeTimeOffReasonsSuccessful = false;
+             try
             {
                 this.telemetryClient.TrackTrace($"{Resource.ProcessKronosToShiftsShiftsAsync} start at: {DateTime.UtcNow.ToString("O", CultureInfo.InvariantCulture)}" + " for isRequestFromLogicApp: " + isRequestFromLogicApp);
 
                 // Sync open shifts from kronos to shifts.
                 await this.openShiftController.ProcessOpenShiftsAsync(isRequestFromLogicApp).ConfigureAwait(false);
 
                 this.telemetryClient.TrackTrace($"{Resource.ProcessOpenShiftsAsync} completed from {Resource.ProcessKronosToShiftsShiftsAsync} ");
+            }
+#pragma warning disable CA1031 // Do not catch general exception types
```

Comment:
The comment here should include the **justification** for the  suppression. Describe why it is ok to catch a generic Exception here.

Reply:
The justification is the same for all the instances here, so disable the warning once, at the top of the method, and re-renable it at the end.

Reply:
@aosolis - ack'd - the changes are being made

Reply:
This has been resolved - please see the newest changes